doc/man3/OSSL_PROVIDER.pod

```diff
@@ -47,6 +52,9 @@ OSSL_PROVIDER_unload() unloads the given provider.
 For a provider added with OSSL_PROVIDER_add_builtin(), this simply
 runs its teardown function.
 
+OSSL_PROVIDER_available() checks if a named provider is available
+for use.
```

Comment:
I think this could do with some expansion around the behaviour with fallbacks. Consider for example this code:

````
int defavl = OSSL_PROVIDER_available(NULL, "default");
OSSL_PROVIDER *legacy = OSSL_PROVIDER_load(NULL, "legacy");
EVP_MD *md;

if (defavl) {
    md = EVP_MD_fetch(NULL, "SHA256", NULL);
} else {
    md = EVP_MD_fetch(NULL, "whirlpool", NULL);
}

    /* Do something with md */
````

My understanding from the code in this PR is that `OSSL_PROVIDER_available()` will return true if the provider is marked as a fallback, even if it hasn't actually been activated. So in the example above if no providers have been explicitly loaded then the `OSSL_PROVIDER_available()` call will return true, even though the subsequent load of the legacy provider means it is no longer available. This could be quite confusing.

Reply:
And otherwise, if asked early, the default will look unavailable...

Reply:
The solution may be for this function to trigger the fallback process. Then it will be able to give a deterministic answer.

Reply:
Yes - that makes a lot of sense.

Reply:
Okie, done.  Some refactoring was needed to get this working.